.github/workflows/compile-examples.yml

```diff
@@ -0,0 +1,265 @@
+              - ~/Arduino/libraries/Firmata/examples/SimpleDigitalFirmata
+              - ~/Arduino/libraries/Firmata/examples/SimpleAnalogFirmata
+              - ~/Arduino/libraries/Firmata/examples/ServoFirmata
+              - ~/Arduino/libraries/Firmata/examples/EchoString
+              - ~/Arduino/libraries/Firmata/examples/AnalogFirmata
+              - ~/Arduino/libraries/Firmata/examples/AllInputsFirmata
+              - libraries/SDU/examples
+          # tian board
+          - board:
+              type: "tian"
+            additional-sketch-paths: |
+              - ~/Arduino/libraries/Firmata/examples/StandardFirmataPlus
+              - ~/Arduino/libraries/Firmata/examples/StandardFirmataEthernet
+              - ~/Arduino/libraries/Firmata/examples/StandardFirmata
+              - ~/Arduino/libraries/Firmata/examples/SimpleDigitalFirmata
+              - ~/Arduino/libraries/Firmata/examples/SimpleAnalogFirmata
+              - ~/Arduino/libraries/Firmata/examples/ServoFirmata
+              - ~/Arduino/libraries/Firmata/examples/EchoString
+              - ~/Arduino/libraries/Firmata/examples/AnalogFirmata
+              - ~/Arduino/libraries/Firmata/examples/AllInputsFirmata
+          # MKRGSM1400 board
+          - board:
+              fqbn: "arduino:samd:mkrgsm1400"
+            additional-sketch-paths: |
+              - ~/Arduino/libraries/MKRGSM/examples
+          # MKRNB1500 board
+          - board:
+              fqbn: "arduino:samd:mkrnb1500"
+            additional-sketch-paths: |
+              - ~/Arduino/libraries/MKRNB/examples
+          # MKRWAN board
+          - board:
+              fqbn: '"arduino:samd:mkrwan1300" "arduino:samd:mkrwan1310"'
+            additional-sketch-paths: |
+              - ~/Arduino/libraries/MKRWAN/examples
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      # The source files are in a subfolder of the ArduinoCore-API repository, so it's not possible to clone it directly to the final destination in the core
+      - name: Checkout ArduinoCore-API
+        uses: actions/checkout@v2
+        with:
+          repository: arduino/ArduinoCore-API
+          path: extras/ArduinoCore-API
+
+      - name: Install ArduinoCore-API
+        run: mv "$GITHUB_WORKSPACE/extras/ArduinoCore-API/api" "$GITHUB_WORKSPACE/cores/arduino"
+
+      - name: Checkout Basic examples
+        uses: actions/checkout@v2
+        with:
+          repository: arduino/arduino-examples
+          path: extras
+
+      - name: Compile examples
+        uses: arduino/compile-sketches@main
+        with:
+          fqbn: ${{ matrix.board.fqbn }}
+          libraries: |
+            - source-url: https://github.com/giulcioffi/WiFi101.git
+              version: namespaced_api
+            - name: MFRC522
+            - name: Arduino_MKRMEM
+            - name: FlashStorage
+            - source-url: https://github.com/arduino-libraries/Keyboard.git
```

Comment:
I'm guess that the use of the development version of the Keyboard library is because that version has an example sketch added:
https://github.com/arduino-libraries/Keyboard/commit/a587425e90590e9e82af055c3c0f292a5e6d70ac

but that example sketch is just a copy/paste of the KeyboardSerial example in the built-in examples:
https://github.com/arduino/arduino-examples/blob/main/examples/09.USB/Keyboard/KeyboardSerial/KeyboardSerial.ino
which the workflow is already configured to compile.

So there would be no loss of coverage from using the release version of the library and removing the `~/Arduino/libraries/Keyboard/examples/Serial` path from `env.UNIVERSAL_SKETCH_PATHS`